airflow/models/dag.py

```diff
@@ -546,6 +553,9 @@ def __init__(
         self.last_loaded = timezone.utcnow()
         self.safe_dag_id = dag_id.replace(".", "__dot__")
         self.max_active_runs = max_active_runs
+        if self.timetable.limit_active_runs is not None:
+            if self.timetable.limit_active_runs < self.max_active_runs:
+                self.max_active_runs = self.timetable.limit_active_runs
```

Comment:
Do you think its worth logging a warning when we invisibly override the DAG config like this?

Reply:
Personally I feel it would be reasonable to fail the DAG since both the timetable and `max_active_runs` are DAG-level arguments and they can be resolved quite easily by the user. Emitting a DagWarning is reasonable too if an error is too aggressive.

Reply:
I'm fine with an error - that way we won't have any confusing mismatch between the code and the actual behaviour of a running DAG.